packages/block-library/src/block/index.php

```diff
@@ -41,14 +42,74 @@ function render_block_core_block( $attributes ) {
 
 	$seen_refs[ $attributes['ref'] ] = true;
 
+	// We need to determine the widest alignment of inner blocks so it can be
+	// applied to the pattern's wrapper.
+	$widest_alignment  = null;
+	$filter_alignments = static function( $parsed_block, $source_block, $parent_block ) use ( &$widest_alignment ) {
+		// If this isn't a top level block in the pattern or we have already
+		// determined that we have a full aligned block, skip it.
+		if ( isset( $parent_block ) || 'full' === $widest_alignment ) {
+			return $parsed_block;
+		}
+
+		$alignment       = _wp_array_get( $parsed_block, array( 'attrs', 'align' ), null );
+		$full_alignments = array( 'full', 'left', 'right' );
+
+		if ( in_array( $alignment, $full_alignments, true ) ) {
+			$widest_alignment = 'full';
+			return $parsed_block;
+		}
+
+		if ( 'wide' === $alignment ) {
+			$widest_alignment = $alignment;
+		}
+
+		return $parsed_block;
+	};
+
+	add_filter( 'render_block_data', $filter_alignments, 10, 3 );
+
 	// Handle embeds for reusable blocks.
 	global $wp_embed;
+
 	$content = $wp_embed->run_shortcode( $reusable_block->post_content );
 	$content = $wp_embed->autoembed( $content );
-
 	$content = do_blocks( $content );
+
 	unset( $seen_refs[ $attributes['ref'] ] );
-	return $content;
+
+	remove_filter( 'render_block_data', $filter_alignments, 10 );
+
+	// Older block versions used only the post's content without incorporating
+	// the editor's wrapper. Newer versions added a wrapper through saved
+	// markup, so utilize it to return only the post's content when appropriate.
+	if ( ! $block_content ) {
+		return $content;
+	}
```

Comment:
Tiny non-blocking thought: when `$block_content` is empty, we also don't need to add or remove the `render_block_data` filter because `$widest_alignment` won't be used for anything.

Reply:
✅  Done (https://github.com/WordPress/gutenberg/pull/54289/commits/392f6e0ef4027b39ea45bc482fbb4590955ebc17)